quinn/src/broadcast.rs

```diff
@@ -0,0 +1,40 @@
+use std::task::{Context, Waker};
```

Comment:
I feel like adding this needs a bunch more justification to help me understand what problem we're trying to solve here, why that particular threshold of problems is crossed with the addition of datagrams. I also wonder what benchmarking you have done to back up the need for this mechanism?

Reply:
This solves a resource-boundedness problem, not a performance one, and it's a problem we have otherwise completely avoided by arranging for futures to otherwise have at most one task waiting on them at a time. The alternative would be to add a `!Clone` `DatagramSender` handle to `NewConnection` with a `fn send_datagram(&mut self, ...)` to ensure that only one sender can exist at a time; that felt a bit like complicating the interface for the implementation's benefit to me.

Reply:
Although, `DatagramSender` *would* let us have a neater API along the lines of

```rust
impl DatagramSender {
    async fn wait_ready(&mut self);
    fn send(&mut self, data: Bytes) -> Result<(), Blocked + ConnectionError>
}
```